packages/web-components/src/components/va-combo-box/va-combo-box-library.js

```diff
@@ -571,8 +571,22 @@ const changeElementValue = (el, value = '') => {
   const selectItem = listOptionEl => {
     const { comboBoxEl, selectEl, inputEl } = getComboBoxContext(listOptionEl);
 
+    // Update the select value after selection
     changeElementValue(selectEl, listOptionEl.dataset.value);
-    changeElementValue(inputEl, listOptionEl.textContent);
+
+    // Update the input value after selection
+    // To prevent display error, use different text extraction method for telephone inputs
+    if (comboBox.isInVaInputTelephone) {
+      // Extract just the flag-text portion or preserve the original input value if re-selecting
+      const flagText = listOptionEl.querySelector('.flag-text');
+      if (flagText) {
+        changeElementValue(inputEl, flagText.textContent);
+      }
+    } else {
+      // Standard behavior for non-telephone inputs
+      changeElementValue(inputEl, listOptionEl.textContent);
+    }
+
```

Comment:
> [!note]
> This fixes the re-selection issue in https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/4224. I bundled these two fixes in a single PR because this fix #4223 relied on CSS truncation. Let me know if you would like me to break this out into a follow-on PR.